app/src/main/java/com/dmuench/scatterhunt/CreateGoalFragment.java

```diff
@@ -71,21 +72,33 @@ public void onCompletedForm() {
         Log.i("FORM CLUE ONE", clueStepOne.getStepData());
         Log.i("FORM CLUE TWO", clueStepTwo.getStepData());
         Log.i("FORM CLUE THREE", clueStepThree.getStepData());
-        Log.i("LATITUDE", Double.toString(state.getDouble("latitude")));
-        Log.i("LONGITUDE", Double.toString(state.getDouble("longitude")));
+        Log.i("FORM LATITUDE", state.getString("latitude"));
+        Log.i("FORM LONGITUDE", state.getString("longitude"));
 
-        String[] clues = new String[]{clueStepOne.getStepData(),  clueStepTwo.getStepData(), clueStepThree.getStepData()};
-        Goal goal = new Goal(titleStep.getStepData(), state.getDouble("latitude"),state.getDouble("longitude"),5, clues , FirebaseAuth.getInstance().getCurrentUser());
-        FirebaseFirestore db = FirebaseFirestore.getInstance();
-        db.collection("Goals").add(goal);
+        String latitude, longitude;
+        latitude = state.getString("latitude");
+        longitude = state.getString("longitude");
 
+        if (latitude == null || longitude == null)
+            Toast.makeText(getContext(), "Location Data Not Available - Please Ensure Location Is Enabled", Toast.LENGTH_SHORT).show();
+        else {
 
-        // TODO: Do Some Firebase Firestore Things.
+            String[] clues = new String[]{clueStepOne.getStepData(), clueStepTwo.getStepData(), clueStepThree.getStepData()};
+            Goal goal = new Goal(titleStep.getStepData(), Double.parseDouble(latitude), Double.parseDouble(longitude), 5, clues, FirebaseAuth.getInstance().getCurrentUser());
+            FirebaseFirestore db = FirebaseFirestore.getInstance();
+            db.collection("Goals").add(goal);
+
+            Navigation.findNavController(getView()).navigate(R.id.returnToMainFragment);
+
+
+
+
+//             TODO: Do Some Firebase Firestore Things. DONE
```

Comment:
If a TODO is DONE, you should remove the comment entirely.

Reply:
removed